mockSchemaRegistryClient_test.go

```diff
@@ -1,146 +1,595 @@
-	schemaReg3, _ := srClient.GetSchema(3)
-	require.JSONEq(t, schema2, schemaReg3.schema)
-	assert.Equal(t, 2, schemaReg3.version)
-	schemaReg4, _ := srClient.GetSchema(4)
-	require.JSONEq(t, schema2, schemaReg4.schema)
-	assert.Equal(t, 2, schemaReg4.version)
-	schemaReg5, _ := srClient.GetSchema(5)
-	require.JSONEq(t, schema3, schemaReg5.schema)
-	assert.Equal(t, 1, schemaReg5.version)
-	schemaReg6, _ := srClient.GetSchema(6)
-	require.JSONEq(t, schema4, schemaReg6.schema)
-	assert.Equal(t, 2, schemaReg6.version)
-
-	// Test registering already registered schema
-	_, err := srClient.CreateSchema("test1-key", schema, Avro)
-	assert.EqualError(t, err, "POST \"mock://testingUrl/subjects/test1-key/versions\": Schema already registered with id 2")
-
-	// Test registering already registered schema
-	_, err = srClient.CreateSchema("test1_arb", schema3, Avro)
-	assert.EqualError(t, err, "POST \"mock://testingUrl/subjects/test1_arb/versions\": Schema already registered with id 5")
-}
-
-func TestMockSchemaRegistryClient_GetLatestSchema(t *testing.T) {
-
-	latest, err := srClient.GetLatestSchema("test1-key")
-	if err != nil {
-		fmt.Println(err.Error())
-		t.Fail()
-	} else {
-		require.JSONEq(t, schema2, latest.schema)
-	}
-
-	latest, err = srClient.GetLatestSchema("test1_arb")
-	if err != nil {
-		fmt.Println(err.Error())
-		t.Fail()
-	} else {
-		require.JSONEq(t, schema4, latest.schema)
-	}
-}
-
-func TestMockSchemaRegistryClient_GetLatestSchema_WhenNoSchemaRegistered(t *testing.T) {
-	localClient := CreateMockSchemaRegistryClient("")
-	var err error
-	assert.NotPanics(t, func() {
-		_, err = localClient.GetLatestSchema("some-innexistent-schema-subject")
-	})
-	assert.EqualError(t, err, "Subject not found")
-}
-
-func TestMockSchemaRegistryClient_GetSchemaVersions(t *testing.T) {
-	versions, _ := srClient.GetSchemaVersions("test1-key")
-	assert.Equal(t, 2, len(versions))
-
-	versions, _ = srClient.GetSchemaVersions("test1_arb")
-	assert.Equal(t, 2, len(versions))
-}
-
-func TestMockSchemaRegistryClient_GetSchemaByVersion(t *testing.T) {
-	oldVersion, _ := srClient.GetSchemaByVersion("test1-value", 1)
-	require.JSONEq(t, schema, oldVersion.schema)
-
-	oldVersion, _ = srClient.GetSchemaByVersion("test1_arb", 1)
-	require.JSONEq(t, schema3, oldVersion.schema)
-}
-
-func TestMockSchemaRegistryClient_GetSubjects(t *testing.T) {
-	allSubjects, _ := srClient.GetSubjects()
-	sort.Strings(allSubjects)
-	assert.Equal(t, allSubjects, []string{"test1-key", "test1-value", "test1_arb"})
-}
-
-func TestMockSchemaRegistryClient_DeleteSubjectByVersion(t *testing.T) {
-	originalCount, _ := srClient.GetSchemaVersions("test1-value")
-	err := srClient.DeleteSubjectByVersion("test1-value", 1, true)
-	assert.NoError(t, err)
-	newCount, _ := srClient.GetSchemaVersions("test1-value")
-	assert.Equal(t, len(originalCount)-1, len(newCount))
+var avroType = Avro
+var protobuf = Protobuf
+
+func TestMockSchemaRegistryClient_CreateSchema_RegistersSchemaCorrectly(t *testing.T) {
```

Comment:
Is it possible to reduce code repetition in schema definitions?

Reply:
Good one! Update :+1: